File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Java/booking/Seat.java
package booking;

import utils.SeatClass;
import utils.SeatStatus;

public class Seat {
    private String seatNumber;
    private SeatClass seatClass;
    private SeatStatus seatStatus;

    public Seat(String seatNumber, SeatClass seatClass) {
        this.seatNumber = seatNumber;
        this.seatClass = seatClass;
    }

    public void book() {
        this.seatStatus = SeatStatus.BOOKED;
    }

    public String getSeatNumber() {
        return seatNumber;
    }

    public SeatClass getSeatClass() {
        return seatClass;
    }

    public SeatStatus getSeatStatus() {
        return seatStatus;
    }
}

File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Cpp/booking/src/Seat.cpp
#include "../include/Seat.h"

Seat::Seat(std::string seatNumber, SeatClass seatClass)
    : seatNumber(std::move(seatNumber)), seatClass(seatClass), seatStatus(SeatStatus::AVAILABLE) { 
}

void Seat::book() {
    seatStatus = SeatStatus::BOOKED;
}

const std::string Seat::getSeatNumber() const {
    return seatNumber;
}

const SeatClass Seat::getSeatClass() const {
    return seatClass;
}

const SeatStatus Seat::getSeatStatus() const {
    return seatStatus;
}